librz/core/rop.c

```diff
@@ -174,27 +189,56 @@ static bool rz_rop_print_standard_mode(const RzCore *core, const RzCoreAsmHit *h
 	const char *comment = rop_comments ? rz_meta_get_string(core->analysis, RZ_META_TYPE_COMMENT, hit->addr) : NULL;
 	char *asm_op_hex = rz_asm_op_get_hex(asmop);
 	const bool colorize = rz_config_get_i(core->config, "scr.color");
+	bool ret_val = context->ret_val;
 	if (colorize) {
 		RzStrBuf *bw_str = rz_strbuf_new(rz_asm_op_get_asm(asmop));
 		RzAsmParseParam *param = rz_asm_get_parse_param(core->analysis->reg, aop.type);
 		RzStrBuf *colored_asm = rz_asm_colorize_asm_str(bw_str, core->print, param, asmop->asm_toks);
 		rz_asm_parse_param_free(param);
 		if (comment) {
-			rz_cons_printf("  0x%08" PFMT64x " %18s  %s%s ; %s\n",
-				hit->addr, asm_op_hex, colored_asm ? rz_strbuf_get(colored_asm) : "", Color_RESET, comment);
+			if (ret_val) {
+				char *str_buf = rz_str_newf("  0x%08" PFMT64x " %18s  %s%s ; %s\n",
+					hit->addr, asm_op_hex, colored_asm ? rz_strbuf_get(colored_asm) : "", Color_RESET, comment);
+				rz_strbuf_append(context->buf, str_buf);
+				free(str_buf);
+			} else {
+				rz_cons_printf("  0x%08" PFMT64x " %18s  %s%s ; %s\n",
+					hit->addr, asm_op_hex, colored_asm ? rz_strbuf_get(colored_asm) : "", Color_RESET, comment);
+			}
 		} else {
-			rz_cons_printf("  0x%08" PFMT64x " %18s  %s%s\n",
-				hit->addr, asm_op_hex, colored_asm ? rz_strbuf_get(colored_asm) : "", Color_RESET);
+			if (ret_val) {
+				char *str_buf = rz_str_newf("  0x%08" PFMT64x " %18s  %s%s\n",
+					hit->addr, asm_op_hex, colored_asm ? rz_strbuf_get(colored_asm) : "", Color_RESET);
+				rz_strbuf_append(context->buf, str_buf);
+				free(str_buf);
+			} else {
+				rz_cons_printf("  0x%08" PFMT64x " %18s  %s%s\n",
+					hit->addr, asm_op_hex, colored_asm ? rz_strbuf_get(colored_asm) : "", Color_RESET);
```

Comment:
same issue here: seems that you always inject `Color_RESET` if it is not colored. this might be a problem